README.md
<!--
 * @Author: xiaqijian
 * @Date: 2023-03-07 09:11:06
 * @LastEditTime: 2023-03-13 15:18:53
 * @Description: 请填写简介
-->

说明文件

左边数据来源

字段：meetDetailList   设置的方法：setMeetDetailList

左边小圆桌和左边数据一样

选中：curMeet    设置方法：setCurMeet


设置token，关键字段：dispWebToken

Loading    changeLoading

docs/CHANGELOG.md
<!--
 * @File: changelog
 * @Author: liulian
 * @Date: 2019-11-20 19:11:49
 * @version: V0.0.0.1
 * @LastEditTime: 2022-10-10 14:35:01
-->

# 前端开发规范：CHANGELOG规范
## 示例模板

```md
# Changelog
此项目的所有显着更改都将记录在此文件中，如果项目更新相关的信息，请在这里写清楚。

- [项目禅道地址](http://192.168.103.101/zentao/bug-browse-1.html)
- [烟雾测试问题修复文档](https://c0vdzyjz0e.feishu.cn/docs/doccna8U34RRtsfqjhYFPvikGyb#uEl8hj)
- [自测过程中问题记录修复文档](https://c0vdzyjz0e.feishu.cn/docs/doccndikZqeME0hqDUi2kGdcwRe)

> [fix|add|update|remove]
## v0.0.0.17 / 2022-08-20 / liulian 146B2
### 更新说明
-[fix]加载类型为sub,返回登录页是用location.href

## v0.0.0.16 / 2022-08-20 / liulian 146B1
### 更新说明
-[fix][27294]解决通话记录中最新一条为非通讯录成员，选中其他通话记录，会自动选中第一条的问题
-[update]去掉人员详情中的部门显示只显示职位（特殊处理，打包后还原代码）已还原

## v0.0.0.17 / 2022-05-17 / liulian 146B1
### 更新说明
-[fix]调整头部导航和人员的字体大小

## v0.0.0.16 / 2022-04-08 / liulian 146B1
### 更新说明
-[fix][26468]解决登陆账号无通讯录权限是，部门名称显示undefined的问题

## v0.0.0.15 / 2022-03-18 / liulian （146 146莆田应急）
### 更新说明
-[add]头部菜单栏显示除了根据配置来以外还要根据通讯录的权限列表来

## v0.0.0.14 / 2022-01-14 / liulian （146）
### 更新说明
-[update]解决后台通知报错却没有reason时提示为空的问题

## v0.0.0.13 / 2022-01-06 / liulian （146）
### 更新说明
>150部分bug合入
-[fix][25052]去掉原密码的限制 
-[fix][24918]解决偶现刷新页面janus未初始化成功的问题
-[fix][24929]转接成功or转接失败，3s后关掉转接面板 
-[fix]解决无论切换主副手柄成功还是失败页面是switch按钮都会被切换的问题 
-[update]解决手柄视频呼叫，对方语音接听，手柄面板显示为语音类型，但仍存在最大化按钮的问题 
-[fix]解决手柄呼叫成员A，当前群组为空时，点击 手柄区域进行页面跳转，页面一直显示为loading状态 
-[fix]会议界面提示已离开但是人员状态依然是在会议中的状态
-[fix]自己不能取回自己的限制，只在保持队列中限制了；应该在按钮区域也做限制

## v0.0.0.12 / 2021-12-24 / liulian （146）
### 更新说明
-[fix] 解决组呼通知录制录音文件时手柄挂机录音面板仍然显示正在录制的问题
-[add] 新增前端配置项isMidStatus是否为中性环境
-[update] 视频框上不显示对讲按钮

## v0.0.0.11 / 2021-10-13 / liulian  （145_B1） 发布版本
### 更新说明
-[updata]一号通呼叫时，默认号码在第一个，之后按照 用户详情中的号码顺序排列（手机、电话、桌面终端、移动终端）

## v0.0.0.10 / 2021-09-24 / liulian  （145_B1）
### 更新说明
-[fix]调整视频窗口和呼入弹框的位置，保证弹框覆盖视频窗口
-[fix]默认会场不显示会场号e
-[fix]解决视频呼入调度与手柄通话，另一个视频话机呼入，调度员应答新呼叫，与新呼叫建立通话，之前的呼叫被保持。但是页面上手柄当前通话成员与视频取流还是被保持的用户，没有切换成新用户的问题
-[add]新增取回转接功能（外线号码）
-[update]发起一号通操作相当于发起呼叫，发起呼叫后/通话中设为灰色不可用
-[update]屏蔽自己取回自己手柄的这种异常操作，取回按钮置为灰色不可点击状态
-[update]点击手柄通话号码，优先在本群组中搜索(只搜索当前页)，然后再去通讯录中搜索
-[update]会议录音和通话录音保持交互一致性
-[update]呼入队列的时候，提高呼入队列的层级

## v0.0.0.9 / 2021-08-31 / liulian
### 更新说明
-[fix]组呼/广播结束时爆错undefind
-[fix]点名结束时爆错undefined  (2021080519)

## v0.0.0.8 / 2021-07-23 / liulian
### 更新说明
- [fix][23105]解决网页调度视频呼叫简讯点击视频弹窗挂断按钮未挂断通话的问题
- [fix][23107]解决网页调度页面关闭组呼通知弹窗后新建录音文件弹窗未关闭的问题
- [fix][23108]解决锁定会场时，结束会议，提示会场已锁定请先解锁，但页面上成员显示已经离开会场的问题
- [fix][23109]解决视频通话窗口，通话时长文字与其他文字重叠的问题

## v0.0.0.7 / 2021-07-02 / liulian
### 更新说明
- [fix][22848]解决模式3下视频呼叫切换之后，再切换回来视频无法正常显示；（模式3下切换 销毁Janus提升性能）;
- [update]在不开启视频通话的情况下，进行视频呼叫时 在呼叫面板上不显示最大化的按钮

## v0.0.0.6 / 2021-05-20 / liulian
### 更新说明
- [fix][22156]解决未选择快捷组或部门时，进行群组操作，弹窗异常的问题；
- [fix][22175]解决呼叫非通讯录成员，点击手柄区域，提醒异常的问题；
- [fix][22166]解决多次保持/取回后异常，无法正常取回通话的问题；
- [fix][22174]调整轮询和一号通传参，轮询：false, 一号通：true;
- [fix] 解决成功删除群组之后，群组成员并未清空的问题；
- [fix] 主手柄呼叫成功，切换到副手柄通话，手柄呼叫面板显示问题的逻辑调整；
- [update] 当处于视频通话时，点击视频窗上的挂断按钮，先进行关闭视频窗的操作，再进行挂断操作
 
## v0.0.0.5 / 2021-04-29 / liulian
### 更新说明
- [fix][21207] 解决当群组较多时，点击编辑下方没有增加群组按钮的问题
- [update] 解决当当前通话的人员不在通讯录中，点击操作面板跳转是一直显示loading状态
- [update] 解决在调度中嵌入视频调度时总是有两条消息提示的问题（模式3下）

## v0.0.0.4 / 2021-04-06 / liulian
### 更新说明
- [fix][20919] 解决网页调度左侧群组无法上下活动查看全部群组的问题，同步调整通讯录列表、会议列表
- [fix][20933] 解决组呼通知记录查看详情中的取消按钮点击没有作用的问题
- [fix][20937] 解决成员较多时，不在当前成员下，点击正在通话中的成员，没有自动跳转显示出正在通话的成员的问题
- [fix][20945] 解决点对点视频呼叫软手柄，建立视频通话双方看不到视频画面的问题
- [fix][20975] 解决议会商中点击被挂断的成员，点击重新呼叫后弹出新的操作弹框的问题
- [fix][20976] 解决网页调度页面修改密码的规则没有数字字母结合，以及位数的限制的问题
- [fix][20978] 解决会议销毁按钮显示有误的问题，调整"撤销成功"提示的显示
- [fix][20991] 解决点击呼入队列的呼入成员，不能切换呼入弹出进行选择应答的问题
- [fix][21019] 解决网页调度群组新增成员，通讯录部门选中全部人员添加，达到上限提示多次的问题
- [add] 支持配置隐藏手机号码中间4位
- [update] 非调度账号免登录加载调度主界面，显示遮罩并提示“当前登录人员无调度权限”
- [update] 修改密码追加确认新密码的输入框
- [update] 号码全匹配缓存（memTel/memTel2/memMobile/memMsgTel/memJkTel）
**二轮bug修复**

## v0.0.0.3 / 2021-02-23 / liulian
### 更新说明
- [fix][20161] 解决对无成员的快捷组点击组呼，弹框确定不消失的问题
- [fix][20174] 成员详情界面点击视频按钮发起视频呼叫，点击号码根据配置发起或音频或视频呼叫
- [fix][20175] 解决由于本机的时间和服务器上的时间存在时间差，通话建立过段时间才开始计时的问题
- [fix][20176] 解决已选中通话的成员，切换到其他组，点击单呼功能下方的当前通话，没有自动跳转到当前通话成员的问题
- [fix][20254] 解决多个成员呼入调度中心，接听某一成员后来电弹框不消失的问题
- [fix][20262] 解决软手柄会议中禁言后，按钮没有自动变为发言按钮的问题
- [fix][20263] 解决会议成员话机上按#键举手发言，自动解除了A的禁言状态，实际A发言并无声音的问题
- [fix][20267] 解决修改默认加载模块为视频调度，登录网页调度会自动跳转到视频调度页面，但是页面显示空白的问题
- [fix][20268] 新增投屏功能
- [fix][20272] 解决会场记录中记录不完整的问题
- [fix][20310] 解决网页调度群组快捷组成员超过20个，只显示了20个成员的问题
- [fix][20178] 解决退出登录时没有用头部所提供的dispatch对象调用退出登录的接口导致操作员已配置主副手柄，网页调度点击切换主副手柄，提示vice-handle is null的问题
- [add] 菜单栏头部追加修改账号密码入口
**一轮bug修复**

## v0.0.0.2 / 2021-02-22 / liulian
### 更新说明
- [fix] 解决群组中新增成员，确定后弹框不消失，且群组成员不会自动刷新的问题
- [fix] 解决成员已配置简讯号码，成员详情不显示简讯号码的问题
- [fix] 解决组呼通知删除语音文件没有二次确认弹框的问题
- [fix] 解决组呼通知记录，选择语音文件的组呼组呼通知记录，点击转发，没有跳转到转发页面的问题
- [fix] 解决组呼通知记录，选择接受状态失败的成员，点击一键重发，还是会向该条记录所有成员发送组呼通知的问题
- [fix] 解决新建会议-会场号没有标识为必填项的问题
- [fix] 解决用户视频呼入调度中心，软手柄接听，弹框视频和会议会商会场显示重叠的问题
- [fix] 解决调度页面选择成员呼叫，成员视频接听，页面探出视频框。鼠标移至视频框中的结束按钮上，上半个按钮没有焦点的问题
- [fix] 解决页面呼叫成员建立通话后，页面点击录音，结束录音。再次点击录音，结束录音，结束通话。右侧通话记录有生成两条相同的通话记录。且点击录音下载只下载了一个录音文件的问题
- [add] 新增黑色皮肤
- [add] 点击主/副手柄，如正在通话的成员在成员列表中，自动选中成员并显示选中信息 
- [add] 新增配置项：是否支持双击呼出（disp.set.doubleClickCall）
- [add] 新增配置项：是否显示成员离线状态（disp.set.show.offline）
- [add] 新增前端配置项：sys-config.js中 devMode：是否开启端日志打印
- [update] 当前操作员 和 其他操作员的默认会场都 不显示 会议时间统计信息
- [update] 组呼通知等直接对组发起操作，若组中成员超过最大选择人数，自动去除多余人员
**烟雾测试bug修复&&研发侧发现问题修复**

## v0.0.0.1 / 2021-10-23 / liulian
### 更新说明
- [update] 新版网页调度提测
- [add] 新增科技风皮肤

### 软件依赖说明
- scooper-core-rest 
- scooper-video (144+)


### 其他说明：无
```

## 模板说明
1. 人员为版本提交人姓名。
2. 更新说明记录按fix、add、update和remove顺序，不要加入其他的。
3. 其他说明中可以写的比较随意
4. 两个版本之间间隔两行



src/Main.js
/*
 * @File: 主入口文件
 * @Author: liulian
 * @Date: 2019-11-20 19:11:49
 * @version: V0.0.0.1
 * @LastEditTime: 2023-03-13 15:20:18
 */
import React, { PureComponent } from "react";
import { Spin, message } from "antd";
import { connect } from "react-redux";
import { apis } from "./util/apis";
import PageRoute from "./router/page-route";
import Loadable from "react-loadable";
import { commonData } from "./config/config";
import { urlParam, getToken, devMode } from "./config/constants";
import { getUrlRealParam } from "./util/method.js";
import {
  changeLoading,
  setConfigData,
  setNavArr,
  setAccPmPermsList,
} from "./reducer/loading-reducer";
import { ConfigProvider } from "antd";
import zhCN from "antd/es/locale/zh_CN";
import qs from "qs";

function Loading(error) {
  if (error) {
    return null;
  }
  return <Spin size="large" className="global-spin1" />;
}
const SysHeader = Loadable({
  loader: () => import("./component/sys-header"),
  loading: Loading,
});
@connect((state) => state.loading, {
  changeLoading,
  setConfigData,
  setNavArr,
  setAccPmPermsList,
})
class NavPage extends PureComponent {
  constructor(props) {
    super(props);
  }
  loadConfig = async () => {
    const promiseArr = [];
    promiseArr.push(apis.disp.config());
    promiseArr.push(apis.core.getAccPmPerms()); // 修改权限列表
    const resultArr = await Promise.all(promiseArr);
    let data = resultArr[0];
    let permsList = resultArr[1];
    if (data) {
      this.props.setConfigData(data);
    }
    if (permsList) {
      this.props.setAccPmPermsList(permsList);
    }

    if (data.set["disp.set.header.title"]) {
      let title = document.getElementsByTagName("title")[0];
      document.title = data.set["disp.set.header.title"];
    }
    if (getUrlRealParam("skin")) {
      window.scooper.configs.skin = getUrlRealParam("skin");
    } else if (data.set["disp.set.skin.default"]) {
      window.scooper.configs.skin = data.set["disp.set.skin.default"];
    } else {
      window.scooper.configs.skin = "light";
    }
    this.defaultRouter(data, permsList);
    this.getCurUrlParams();
  };
  getCurUrlParams() {
    const { pathname } = this.props.location;
    let urlParams = qs.parse(window.location.href.replace("?", "&"));
    if (!urlParams.skin) return;

    Object.keys(urlParams).forEach((item) => {
      if (item.indexOf(pathname) !== -1) {
        delete urlParams[item];
      }
    });
    commonData.setCommonData("urlParams", urlParams);
  }
  /**
   * 格式化路由信息
   */
  defaultRouter = (data, permsList) => {
    if (JSON.stringify(data) !== "{}") {
      sessionStorage.setItem(
        "queryPageSize",
        data.set["disp.set.core.load.size"]
      ); //设置通讯录的每页加载数
      sessionStorage.setItem(
        "defaultTelStatus",
        data.set["disp.set.show.defaultTelStatus"]
      ); //默认号码状态（callst_none）
      sessionStorage.setItem(
        "defaultTelStatusDesc",
        data.set["disp.set.show.defaultTelStatusDesc"]
      ); //默认号码状态描述（空闲）

      let items = data.nav["disp.nav.items"]; // "dispatch,appMsg..."
      let itemsArr = items.split(","); // ["dispatch","appMsg"]
      let configArr = [];
      itemsArr.forEach((element) => {
        if (
          permsList &&
          permsList.length > 0 &&
          permsList.indexOf("/" + element) > -1
        ) {
          let itemTitle = "disp.nav." + element + ".title";
          let itemUrl = "disp.nav." + element + ".url";
          let param = {
            key: element,
            title: data.nav[itemTitle],
            url: data.nav[itemUrl],
          };
          // token
          if (param.url.indexOf("token=") > -1) {
            devMode &&
              console.log(
                param.title + "模块携带token参数，不自动追加登录账号标识！"
              );
          } else if (param.url.indexOf("?") >= 0) {
            param.url = param.url + "&token=" + getToken();
          } else {
            param.url = param.url + "?token=" + getToken();
          }
          // 皮肤
          if (param.url.indexOf("skin") >= 0) {
            devMode &&
              console.log(
                param.title + "模块携带skin参数，不自动追加登录账号标识！"
              );
          } else {
            param.url =
              param.url + "&skin=" + data.set["disp.set.skin.default"];
          }
          configArr.push(param);
        }
      });
      this.props.setNavArr(configArr);
    }
  };

  componentWillMount() {
    this.loadConfig();
    let token = urlParam.token;
    if (token) {
      sessionStorage.setItem("dispWebToken", token);
    }
    if (!sessionStorage.getItem("dispWebToken")) {
      message.error("请先登录！");
      localStorage.clear();
      sessionStorage.clear();
      // 去掉登录校验
      // window.location.href = "/scooper-dispatch-web";
      // this.props.history.push('/login');
    }
  }

  render() {
    const url = this.props.match.url;
    let { configData } = this.props;
    const body = window.document.body;
    body.className = "skin-science";
    return (
      <ConfigProvider locale={zhCN}>
        <Spin
          size="large"
          className={"global-spin2 test"}
          // spinning={this.props.loading}
          spinning={false}
        >
          <PageRoute url={url} />
        </Spin>
      </ConfigProvider>
    );
  }
}

export default NavPage;
